### ISSUE TITLE
ifctester: ids validation report export error

### ISSUE THREAD

**Issue by pippo-82** [IfcOpenShell/IfcOpenShell#4526]
Hi there,
I run an ifc validation against ids file (success as per attached image) and now I'm trying to exporting the report to the various formats (ods, json, html, bcf) but with no luck.
I'm running the code in Jupiter Notebook and I have manually installed the following packages:

- ifctester (in order to create ids files and test ifc files against ids)
- xmlschema (in order to handle xml data)
- bcf-client (in order to export report in bcf format)
- pystache (in order to export report in html format)
- odfpy (in order to export report in ods format)

My sample code:

```python
### using code to create an ids file from here: [ifctester](https://pypi.org/project/ifctester/)
### opening an ifc file
my_ifc = ifcopenshell.open(ifc_filename)
### validating ids
my_ids.validate(my_ifc)
### show results in Console
reporter.Console(my_ids).report() # shows passes/fails as in image
### show report as BCF file
reporter.Bcf(my_ids).report() # shows passes/fails

## TRYING TO EXPORT TO BCF
### define Bcf filepath
sBcfFilepath = ifc_filename.replace("ifc","bcf")
### writing report to bcf file
**reporter.Bcf(my_ids).write(sBcfFilepath) # no error but no file is being created**

## TRYING TO EXPORT TO HTML
### define Html filepath
sHtmlFilepath = ifc_filename.replace("ifc","html")
### save report as HTML file
**reporter.Html(my_ids).to_file(sHtmlFilepath) # a file html is created but it's empty, it seems only the template has been saved**

## TRYING TO EXPORT TO JSON
### define JSON filepath
sJSONFilepath = ifc_filename.replace("ifc","json")
### save report as JSON file
**reporter.Json(my_ids).to_file(sJSONFilepath) # a json file is created but it's empty ( contains only {} )**

## TRYING TO EXPORT TO ODS
### define ODS filepath
sODSFilepath = ifc_filename.replace("ifc","ods")
### save report as ODS file
**reporter.Ods(my_ids).to_file(sODSFilepath) # throws error KeyError: 'title' pointing to line 339 of reporter.py (probably not accessing correctly the ids report title property)**
```

Image with report preview in Jupiter Notebook
![IfcTester_tests](https://github.com/IfcOpenShell/IfcOpenShell/assets/75814336/f53d3983-19f6-492c-9d9a-3c03fdff6127)

Thanks in advance for your kind reply.

Paolo

**Comment by Moult** [comment]
Is this error specific to Jupyter notebooks?

**Comment by pippo-82** [comment]
Hi Dion, thanks for your reply!
I tested only in Jupiter Notebooks for now.
Tomorrow I'll test it in VS Code and revert.
Bye

**Comment by pippo-82** [comment]
Hey @Moult , just for feedback, I tried also in another venv in VS Code (different Python 3.9.13 x64) and I get the same result.
Code as shown above.

Thanks.
Paolo

**Comment by Moult** [comment]
Have you got a sample IDS and IFC to reproduce?

**Comment by pippo-82** [comment]
Hey @Moult ,
the ids was created with the code from idstester pypi (https://pypi.org/project/ifctester/)

The ifc file is the alle Clinic Architecture from here (https://github.com/buildingSMART/Sample-Test-Files/blob/master/IFC%202x3/Medical-Dental%20Clinic/Clinic_Architectural.ifc)

Thanks again. 
Paolo

**Comment by marszulc** [comment]
Hello,
not an expert but from my experiences, it seems you lack .report() function:

```
reporter_obj = reporter.Html(my_ids)
reporter_obj.report()
reporter_obj.to_file(sHtmlFilepath)
```

**Comment by pippo-82** [comment]
> Hello, not an expert but from my experiences, it seems you lack .report() function:
> 
> ```
> reporter_obj = reporter.Html(my_ids)
> reporter_obj.report()
> reporter_obj.to_file(sHtmlFilepath)
> ```

Hey @marszulc ,
sorry for the late reply.
I already tried that, I skipped it in the code transcription above but I used the .report() before trying to export any of the report file formats.

I tested again with python even in Blender 4.1 (just downloaded) with the sample files and I get the same result: the html file opened is empty.

@Moult did you have the chance to replicate the issue?

Thanks
Paolo

**Comment by pippo-82** [comment]
**UPDATE**
Just tested again in Blender with other IFC and IDS files and it works!
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/75814336/9ce9e158-06dd-460d-b99e-745698e72f75)


I used these files now:
[IFC](https://github.com/buildingSMART/IDS/blob/master/Documentation/testcases/property/fail-a_logical_unknown_is_considered_false_and_will_not_pass.ifc)
[IDS](https://github.com/buildingSMART/IDS/blob/master/Documentation/testcases/property/fail-a_logical_unknown_is_considered_false_and_will_not_pass.ids)


The html output
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/75814336/3c8580ac-8515-40bc-9a72-348cf42e2437)

The ods output opened in Excel
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/75814336/603bc9c5-aae3-4803-94b9-1e8e8842a8af)

In this case there is no geometry, the export to BCF throws an error.
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/75814336/c69d264e-e7fe-437f-b465-0ca0ca6de9f1)

I'll try again in Python now, firstly in Blender, then in other venv.

**Comment by pippo-82** [comment]
OK guys, I figured it out! 🤦‍♂️
It was my oversight I didn't instantiate the report object as @marszulc advised.
I solved in Python within Blender 4.1 for now.
See my last post on LinkedIn.

Still getting the error on BCF export though (see last picture in my previous post.
@Moult do you know what may means by any chance?

Thanks a lot guys!
Paolo

**Comment by astridvdl** [comment]
When writing an .ods report, the report generated has an error when opened in excel. This error persists whether using Bonasi/Blender or ifcopenshell and ifctester in a Python script.

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><logFileName>error242520_01.xml</logFileName><summary>Errors were detected in file '[FILE NAME REDACTED]'</summary><repairedRecords><repairedRecord>Repaired Records: Worksheet properties from /xl/workbook.xml part (Workbook)</repairedRecord></repairedRecords></recoveryLog>`

From what I have read, it might be the specification names, some are over 30 characters which is apparently not ideal for sheet names. Might anyone have a solution to this? If I open the file in excel and resave it, the error goes away. However I am attempting an automated workflow and do not want to open these reports individually.
